unyt/array.py

```diff
@@ -131,7 +131,14 @@
 
 NULL_UNIT = Unit()
 POWER_MAPPING = {multiply: lambda x: x, divide: lambda x: 2 - x}
-DISALLOWED_DTYPES = ("S", "U", "a", "O", "M", "m", "b")
+DISALLOWED_DTYPES = (
+    "S",  # string
+    "U",  # (unicode) bytes
+    "a",  # bytestring
```

Comment:
```suggestion
    "S",  # bytestring
    "a",  # bytestring
    "U",  # (unicode) bytes
```

Small fix, both "a" and "S" are bytestring.